docs/data/data-grid/editing/editing.md

```diff
@@ -4,266 +4,411 @@ title: Data Grid - Editing
+}
+```
 
-Having all cells of a row in edit mode allows validating a field based on the value of another one.
-To accomplish that, start by adding the `preProcessEditCellProps` as explained in the [cell editing](#client-side-validation).
-When the callback is called, use the API to check the value of the other field and decide if the current value is valid or not.
-Return a new object contaning `params.props` and the `error` attribute with the validation status.
-Once at the least one field has the `error` attribute equals to true no new value will be committed.
+### With auto-stop
 
-**Note:** For server-side validation, the same [approach](#server-side-validation) from the cell editing can be used. The data grid will wait for all promises to resolve before commiting.
+An edit component has an "auto-stop" behavior when it stops the edit mode as soon as the value is changed.
+To picture better, imagine an edit component with a combo, created following the steps above.
+By default, it would require two clicks to change the value of the cell: one click to select a new value and another click outside to save.
+This second click can be avoided if the first click also stops the edit mode.
+To create an edit component with auto-stop, call `apiRef.current.stopCellEditMode` after setting the new value.
+Since `apiRef.current.setEditCellValue` may do additional processing, it is necessary to first wait for it to resolve before stopping the edit mode.
+Also, it is a good practice to check if `apiRef.current.setEditCellValue` returned `true`.
+It will be `false` if `preProcessEditProps` set an error during [validation](#validation).
 
-The following demo requires a value for the "Payment method" column if the "Is paid?" column was checked.
+```tsx
+const handleChange = async (event: SelectChangeEvent) => {
+  const isValid = await apiRef.current.setEditCellValue({
+    id,
+    field,
+    value: event.target.value,
+  });
+
+  if (isValid) {
+    apiRef.current.stopCellEditMode({ id, field });
+  }
+};
+```
 
-{{"demo": "ConditionalValidationGrid.js", "disableAd": true, "bg": "inline", "defaultCodeOpen": false}}
+The demo below implements an edit component with auto-stop, based on a native [Select](https://mui.com/components/selects), for the Role column.
 
-> The conditional validation can also be implemented with the [controlled editing](#controlled-editing-2).
-> This approach can be used in the free version of the DataGrid.
-> The only limitation is that it does not work with the `singleSelect` column type.
+{{"demo": "AutoStopEditComponent.js", "bg": "inline", "defaultCodeOpen": false}}
 
-### Control with external buttons [<span class="plan-pro"></span>](https://mui.com/store/items/material-ui-pro/)
+> ⚠ We don't recommend using edit components with auto-stop in columns that use long-running `preProcessEditCellProps` because the UI will freeze while waiting for `apiRef.current.setEditCellValue`.
```

Comment:
```suggestion
> ⚠ **Note:** We don't recommend using edit components with auto-stop in columns that use long-running `preProcessEditCellProps` because the UI will freeze while waiting for `apiRef.current.setEditCellValue`.
```